### PR TITLE
(Feature) Added way to output metrics for pytorchlightning during training

### LINKED ISSUE #600: Add more logging to the pytorch lighning models.
The feature consists of two parts:
  1. Add the validation loss to the progress bar by default
  2. Create boolean parameter for extended progress bar logging (showing the different components of the loss)

### PR DESCRIPTION
**Pull request by fstroth**
With this PR the `LightningModelAdapter` will take another `__init__` argument named `metrics_keys_to_log_to_prog_bar` which has to be a list of string tuples with two elements, were the first entry is the name of the metric to log and the second one is the display name.
By default the value will be set to `[("AP (IoU=0.50:0.95) area=all", "mAP")]` which will log the mAP. This will only log the value is the COCOMetric is supplied as a metric! If so it will look like this:

image

Closes #600

### CODE REVIEW

**Comment by fstroth** [comment]
The test seem to fail because `pytest-mock` is not installed for the test environment. Where should I add the dependency?

**Comment by ai-fast-track** [comment]
That's great @fstroth!

I think it would be good to use `COCOMetric` as the default title to be consistent with our current Fastai logging:
instead of: 
`[("AP (IoU=0.50:0.95) area=all", "mAP")]`
We would use this:
`[("AP (IoU=0.50:0.95) area=all", "COCOMetric")]

image

**Comment by ai-fast-track** [comment]
> The test seem to fail because `pytest-mock` is not installed for the test environment. Where should I add the dependency?

In the `setup.cfg`  `[dev]` section?
https://github.com/airctic/icevision/blob/master/setup.cfg#L56

**Comment by fstroth** [comment]
@ai-fast-track Yes the test is passing locally, all other tests also pass except the yolo ones because I didn't install the yolo part. Training a model with the code also worked (that's where the screenshot above is from).
